### PR TITLE
Enforce coding style on misc Ui files

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This is quite a big one as Window and Widget are used in lots of locations. I've moved widget into the Widget.h header as well.

### CODE REVIEW

File: src/OpenLoco/Widget.cpp [hunk 1]
```diff
@@ -11,40 +11,36 @@ using namespace OpenLoco::Interop;
 
 namespace OpenLoco::Ui
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Widget.cpp:12]: You've changed this to repeat `Widget::` in function signatures. Why not just do:
```suggestion
namespace OpenLoco::Ui::Widget
```

**Comment by AaronVanGeffen** [comment]
I'm not sure about moving the widget draw functions to be member functions of the Widget struct. I'd rather their visibility remain more limited.

**Comment by duncanspumpkin** [comment]
Hmm well i'll have to move them to a different namespace then as Ui::Widget is now for the type.

**Comment by duncanspumpkin** [comment]
Okay ive made them private and left a note to go back and make them normal member functions.

File: src/OpenLoco/Window.h [hunk 2]
```diff
@@ -83,102 +62,15 @@ namespace OpenLoco::Ui
...
-    constexpr widget_t widgetEnd()
-    {
-        widget_t out = {};
-        out.type = widget_type::end;
-
-        return out;
+        constexpr uint8_t none = 0;
+        constexpr uint8_t horizontal = (1 << 0);
+        constexpr uint8_t vertical = (1 << 1);
+        constexpr uint8_t both = (1 << 0) | (1 << 1);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Window.h:70]: Now that this is a namespace, we could just write:
```suggestion
        constexpr uint8_t both = horizontal | vertical;
```

**Review by AaronVanGeffen — APPROVED** [review]
The new type names are a solid improvement. Code changes look good as well. Haven't tested the changes though.

**Pull request merged**